src/java/main/org/apache/zookeeper/server/auth/WrappedAuthenticationProvider.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zookeeper.server.auth;
+
+import org.apache.zookeeper.KeeperException;
+import org.apache.zookeeper.data.ACL;
+import org.apache.zookeeper.server.ServerCnxn;
+import org.apache.zookeeper.server.ZooKeeperServer;
+
+import java.util.List;
+
+/**
+ * Provides backwards compatibility between older {@link AuthenticationProvider}
+ * implementations and the new {@link ServerAuthenticationProvider} interface.
+ */
+class WrappedAuthenticationProvider extends ServerAuthenticationProvider {
+    private final AuthenticationProvider implementation;
+
+    static ServerAuthenticationProvider wrap(AuthenticationProvider provider) {
+        return (provider instanceof ServerAuthenticationProvider) ? (ServerAuthenticationProvider)provider
+                : new WrappedAuthenticationProvider(provider);
+    }
+
+    private WrappedAuthenticationProvider(AuthenticationProvider implementation) {
+        this.implementation = implementation;
+    }
+
+    @Override
+    public KeeperException.Code handleAuthentication(ZooKeeperServer zks, ServerCnxn cnxn, byte[] authData) {
+        return implementation.handleAuthentication(cnxn, authData);
+    }
+
+    /**
+     * {@inheritDoc}
```

Comment:
I've never used this before and from the name it implies that it will copy the doc blurb from the super class when running the javadoc tool. Is this correct? Do we have to do it for this method only or for others as well?